metrics/firebase/functions/test/main_test.dart

```diff
@@ -450,6 +454,441 @@ void main() {
       },
     );
   });
+
+  group("onBuildUpdatedHandler", () {
+    final newDocumentSnapshot = DocumentSnapshotMock();
+    final change = ChangeMock();
+    final documentQuery = DocumentQueryMock();
+    final querySnapshot = QuerySnapshotMock();
+    final oldBuildJson = testDataGenerator.generateBuildJson(
+      buildStatus: BuildStatus.inProgress,
+      duration: null,
+    );
+    final oldBuildDocumentData = DocumentData.fromMap(oldBuildJson);
+
+    PostExpectation<Future<QuerySnapshot>> whenGetTaskDocument() {
+      whenFirestore().thenReturn(firestore);
+      when(firestore.collection(tasksCollectionName))
+          .thenReturn(collectionReference);
+      when(collectionReference.where(
+        any,
+        isEqualTo: anyNamed('isEqualTo'),
+      )).thenReturn(documentQuery);
+
+      return when(documentQuery.get());
+    }
+
+    PostExpectation<List<DocumentSnapshot>> whenTaskDocuments() {
+      whenGetTaskDocument().thenAnswer((_) => Future.value(querySnapshot));
+
+      return when(querySnapshot.documents);
+    }
+
+    PostExpectation<DocumentData> whenAccessChangeData({
```

Comment:
I think it would be more clear and SRP to make 2 different methods like `whenChangeBeforeData` and `whenChangeAfterData`.